### PR TITLE
FreeLdr: Manage cmdline and FREELOADER section as global settings; don't popup about any deprecated features if autoboot.

### PR DESCRIPTION
**Pull request by HBelusca**
## Purpose

Move management of basic FreeLdr "global" settings, obtained either from the multiboot command line or the `FREELOADER` freeldr.ini section, to a distinct file, and make these global settings visible globally: no need to use helper functions to access them.

In addition: following commit 5f3554a40, don't popup about deprecated features in case of autoboot (booting without user interaction).

## Proposed changes

```
[FREELDR] The multiboot command-line and FREELOADER section options become global settings

Dual-license settings.c: "old" code (CmdLineParse) stays BSD
and new code (mine) becomes MIT.
```

```
[FREELDR] Don't popup about deprecated features in auto-boot scenario

Addendum to commit 5f3554a40.
```

### CODE REVIEW

File: boot/freeldr/freeldr/settings.c [hunk 1]
```diff
@@ -1,9 +1,10 @@
 /*
- * PROJECT:         ReactOS Boot Loader
- * LICENSE:         BSD - See COPYING.ARM in the top level directory
- * FILE:            boot/freeldr/freeldr/cmdline.c
- * PURPOSE:         FreeLDR Command Line Parsing
- * PROGRAMMERS:     ReactOS Portable Systems Group
+ * PROJECT:     FreeLoader
+ * LICENSE:     BSD - See COPYING.ARM in the top level directory
+ *           or MIT (https://spdx.org/licenses/MIT)
```
Comment (binarymaster) [inline boot/freeldr/freeldr/settings.c:4]: It can be both? Looks complicated. 🤔

File: boot/freeldr/freeldr/settings.c [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+ * PURPOSE:     Command-line parsing and global settings management
+ * COPYRIGHT:   Copyright 2008-2010 ReactOS Portable Systems Group
```
Comment (binarymaster) [inline boot/freeldr/freeldr/settings.c:6]:
```suggestion
 * COPYRIGHT:   Copyright 2008-2010 ReactOS Portable Systems Group <ros.arm@reactos.org>
```

File: boot/freeldr/freeldr/settings.c (hunk 1 above)
Comment (HBelusca) [reply boot/freeldr/freeldr/settings.c:4]: I have no idea how to write the header to tell it's "dual licensed" in the usual sense.

**Review by DarkFire01 — APPROVED** [review]
I'm good with it, just fix that COPYING.ARM before merge please!

File: boot/freeldr/freeldr/settings.c (hunk 1 above)
Comment (DarkFire01) [reply boot/freeldr/freeldr/settings.c:4]: Oh, derp. 👍

**Comment by HBelusca** [comment]
> I'm good with it, just fix that COPYING.ARM before merge please!

@DarkFire01 As Stanislav pointed, the file still exists (precisely because we have BSD code with this specific license in our codebase -- it contains the ROS-specific clause), so I guess that unless you guys are OK on how to do the "double-licensing", I can merge sometime this week.

**Comment by JoachimHenze** [comment]
image

**Comment by HBelusca** [comment]
Yes @JoachimHenze they were meant to be separate commits, but they all still refer to the same PR number.

**Pull request closed without merging**